src/keys.rs

```diff
@@ -1288,4 +1295,46 @@ mod tests {
             assert_eq!(internal_ovk.0, tv.internal_ovk);
         }
     }
+
+    #[test]
+    fn bip340_test_vector() {
+        for tv in crate::test_vectors::bip340_vector::test_vectors() {
+            let isk = IssuanceAuthorizingKey::from_bytes(tv.isk).unwrap();
+
+            let ik = IssuanceValidatingKey::from(&isk);
+            assert_eq!(ik.to_bytes(), tv.ik);
+
+            let message = tv.message;
+            let aux_rand = [0u8; 32];
+
+            let signature = SigningKey::from(isk.0)
+                .sign_prehash_with_aux_rand(&message, &aux_rand)
+                .unwrap();
+            let sig_bytes: [u8; 64] = signature.to_bytes();
+            assert_eq!(sig_bytes, tv.signature);
+
+            assert!(VerifyingKey::try_from(ik.0)
+                .unwrap()
+                .verify_prehash(&message, &signature)
+                .is_ok());
+        }
+    }
+
+    #[test]
+    fn zsa_issuance_auth_sig_test_vectors() {
```

Comment:
This test uses the test vectors generated via the python code in [this PR](https://github.com/QED-it/zcash-test-vectors/pull/12).